bagit.py

```diff
@@ -761,14 +808,10 @@ def validate_fetch(self):
         Raises `BagError` for errors and otherwise returns no value
         """
 
-        for url, file_size, filename in self.fetch_entries():
-            # fetch_entries will raise a BagError for unsafe filenames
-            # so at this point we will check only that the URL is minimally
-            # well formed:
-            parsed_url = urlparse(url)
-
-            if not all((parsed_url.scheme, parsed_url.netloc)):
-                raise BagError(_("Malformed URL in fetch.txt: %s") % url)
```

Comment:
I think we should still have this validation step since it reports some malformed URLs earlier. This would also allow us to remove both of the `fnmatch` checks by changing the whitelist to be a list of URL schemes and simply adding a second `if parsed_url.scheme not in self.fetch_url_schema_whitelist: …` check here.